### PR TITLE
gov/proposal: Correct proposals remaining time

### PR DESCRIPTION
**Pull request by dmigwi**
Fix the proposal remaining time value on the proposal details page.

### CODE REVIEW

**Comment by dmigwi** [comment]
#1500

File: gov/politeia/types/types.go [hunk 1]
```diff
@@ -240,12 +240,12 @@ type ProposalMetadata struct {
...
 // and a target block time for the network, so those must be provided as
 // arguments.
-func (pinfo *ProposalInfo) Metatdata(tip, targetBlockTime int64) *ProposalMetadata {
+func (pinfo *ProposalInfo) Metadata(tip, targetBlockTime int64) *ProposalMetadata {
 	meta := new(ProposalMetadata)
 	desc := pinfo.VoteStatus.ShortDesc()
 	switch desc {
 	case "Authorized":
-		blocksTil := windowSize - tip%windowSize
+		blocksTil := int64(windowSize)
```
Comment (buck54321) [inline gov/politeia/types/types.go]: You're correct that this was wrong, but I'm not sure this fixes it. Actually, I'm not sure we can know the `StartHeight` on an `"Authorized"` proposal. It starts whenever the admin approves. That was my mistake.

File: gov/politeia/types/types.go [hunk 2]
```diff
@@ -246,7 +246,6 @@ func (pinfo *ProposalInfo) Metadata(tip, targetBlockTime int64) *ProposalMetadat
 	switch desc {
 	case "Authorized":
 		blocksTil := int64(windowSize)
-		meta.StartHeight = tip + blocksTil
 		meta.SecondsTil = blocksTil * targetBlockTime
```
Comment (buck54321) [inline gov/politeia/types/types.go]: Can just use `windowSize` now. Don't need `blocksTil`.

File: views/proposal.tmpl [hunk 3]
```diff
@@ -41,8 +41,8 @@
                 <span class="fs18 font-weight-bold lh1rem">Vote scheduled</span>
                 <div class="py-2">
                   {{.Username}} has authorized voting on this proposal.
-                  Voting begins at block {{$metadata.StartHeight}}
-                  (about <span class="medium-sans">{{secondsToShortDurationString $metadata.SecondsTil}}</span>)
+                  Waiting for the Politeia admin to approve the vote start.
+                  (should last for <span class="medium-sans">{{secondsToShortDurationString $metadata.SecondsTil}}</span>)
```
Comment (buck54321) [inline views/proposal.tmpl]: How about "Awaiting admin approval (vote duration: 7 days)" or even just "Awaiting admin approval"? Drop the period as well.

`SecondsTil` is not the right field for this value. It's a constant of 7 days here, so we don't actually need a variable or even to list it.

File: gov/politeia/types/types.go [hunk 4]
```diff
@@ -240,14 +240,12 @@ type ProposalMetadata struct {
...
-func (pinfo *ProposalInfo) Metatdata(tip, targetBlockTime int64) *ProposalMetadata {
+func (pinfo *ProposalInfo) Metadata(tip, targetBlockTime int64) *ProposalMetadata {
 	meta := new(ProposalMetadata)
 	desc := pinfo.VoteStatus.ShortDesc()
 	switch desc {
 	case "Authorized":
-		blocksTil := windowSize - tip%windowSize
-		meta.StartHeight = tip + blocksTil
-		meta.SecondsTil = blocksTil * targetBlockTime
+		meta.SecondsTil = int64(windowSize) * targetBlockTime
```
Comment (buck54321) [inline gov/politeia/types/types.go]: Sorry @dmigwi. I was probably not being very clear here. We have no reason to set this field, and doing so is a little misleading. Seconds until what? This value, as you've set it, is a constant. It's just the size of the window converted to time, and is purposefully set to be 7 days. It is not the time til voting starts or ends or until any other time relevant to the metadata for an authorized proposal.

For the metadata, we expect some fields to be unset during certain stages of the proposal process. You can just drop the "Authorized" case altogether here.

**Pull request merged**